### ISSUE TITLE
benefits of ember-cli-enjection

### ISSUE THREAD

**Issue by antulik** [williamsbdev/ember-cli-injection#4]
Is this the same functionality?

_ember-cli-injection_

``` js
import Ember from 'ember'
import inject from 'ember-cli-injection/inject';

export default Ember.Controller.extend({
  fooRepo: inject('repos')('foo'),

  doSomething: function() {
    this.get('fooRepo').doStuff();
  }
});
```

_standard ember_

``` js
import Ember from 'ember';

export default Ember.Controller.extend({
  fooRepo: Ember.computed(function() {
    return this.container.lookup('repos:foo')
  }),

  doSomething: function() {
    this.get('fooRepo').doStuff();
  }
});
```

**Comment by toranb** [comment]
more or less :)

the benefit is that you can not reference container in your controllers/components (and instead just use the handy inject function to lazily grab a dependency from ember). Services in ember 2.x are similar actually - so if you see foo: Ember.inject.service() it's nearly identical. The big reason we wrote this library is that we have non service objects (that ember isn't aware of) that need injected also.

**Comment by antulik** [comment]
Can you give an example of objects (other than repositories) which wouldn't fit into ember.services?

**Comment by williamsbdev** [comment]
I don't have an example of an object other than a repository. In one of our
apps at work, we have our repositories and then one or two services.

There is nothing stopping these repositories from being services and
eliminating this addon and the auto-register addon. If you like that
approach, by taking of built-in ember functionality, that would be less
customization of your app.

**Comment by toranb** [comment]
@antulik as Mr Williams stated you _could_ make each repository an ember service. When he and I worked w/ this addon a while back it was unclear what role a service would play (ie- how might it be different than other singleton objects in your app/etc).

I think it's your preference - a service plays well here because each repository is a singleton. In our apps we decided to stick w/ repository as a plain ember object because the primary "function" of a repository was very different than my interpretation of what a service does/is in the ember ecosystem today.

A repository (the way you see @williamsbdev and I use it anyway) is stateless. Traditionally ember services are used to share state across several views (or components) in your app. This seemingly subtle difference is worth calling out because when I hear a team member say the word "service" it provides some context (ie- how it's used in our app). I like having a repository mean something different (in our apps). 

Again - preference but the example above is what my team talked about internally and something to consider when you decide :)
